packages/esm-form-entry-app/src/app/form-submission/form-submission.service.ts

```diff
@@ -131,6 +129,47 @@ export class FormSubmissionService {
     }
   }
 
+  private confirmVisitDateAdjustment() {
+    return confirm(
+      'The encounter date falls outside the designated visit date range. Would you like to modify the visit date to accommodate the new encounter date?',
+    );
+  }
+
+  private submitEncounter(encounterCreate: EncounterCreate): Observable<any | undefined> {
+    if (!encounterCreate) {
+      return of(undefined);
+    }
+
+    if (encounterCreate.uuid) {
+      return this.encounterResourceService.getEncounterByUuid(encounterCreate.uuid).pipe(
+        take(1),
+        mergeMap((encounter) => {
+          if (
```

Comment:
Currently, this is always checking encounter date against visit date. Even though encounter date may not have been changed or the POST to update the encounter would return 200 meaning dates are OK. Would it make sense to let the POST go through to update visit 1st and, based on the returned error (400?), check the dates?

Reply:
I think it's preferable to do the check here, since that ensures that we have the same logic in offline mode. Otherwise, an offline form might end up being rejected (then again, I'm a little vague on whether or how we're tracking visits for offline mode)